src/librustc_span/source_map.rs

```diff
@@ -194,12 +209,25 @@ impl SourceMap {
         self.files.borrow().stable_id_to_source_file.get(&stable_id).map(|sf| sf.clone())
     }
 
-    fn next_start_pos(&self) -> usize {
-        match self.files.borrow().source_files.last() {
-            None => 0,
-            // Add one so there is some space between files. This lets us distinguish
-            // positions in the `SourceMap`, even in the presence of zero-length files.
-            Some(last) => last.end_pos.to_usize() + 1,
+    fn allocate_address_space(&self, size: usize) -> Result<usize, OffsetOverflowError> {
+        let size = u32::try_from(size).map_err(|_| OffsetOverflowError)?;
```

Comment:
Just trying to learn here, hope I'm not adding noise to the PR - what is happening here? Is this basically handling the case where `usize` is larger than 32 bits and so `size` is larger than what fits in a `u32`?

Reply:
Yes, exactly. `u32` is always 32bits, `u64` is always 64bits but `usize` is basically the size of a reference/pointer on the target platform. So for 32 bit environments, `usize` will be 32bits and for 64 bit environments, `usize` will be 64bits. So in a 64 bit environment, this conversion might fail because the value of `size` is larger than will fit in a 32 bit integer.

Reply:
Thanks!